processor/intracesampler/README.md

```diff
@@ -0,0 +1,63 @@
+# In-Trace Sampling Processor
+
+| Status                   |                               |
+| ------------------------ |-------------------------------|
+| Stability                | [in development]              |
+| Supported pipeline types | traces                        |
+| Distributions            | [contrib]                     |
+
+Supported pipeline types: traces
+
+The in-trace sampler can sample (remove) individual spans from a complete trace when they are too costly / verbose / noisy.
+
+To function properly, the sampler must receive "complete" traces from which it can derive contextual span properties used for sampling decisions, such as roots and leaves. It means that all spans for a given trace MUST be received by the same collector instance and that in-trace sampler processor MUST be placed in the trace pipeline after a processor that group spans by trace id (such as `tail_sampling` or `groupbytrace`)
+
+Just like any other sampler, the in-trace sampler can be configured such that for each span it can either sample (keeps all spans in a trace and does not change it) or not-sample (remove some of the spans in the trace). The `sampling_percentage` config option is used to control how many traces are passed as is and how many are sampled, so the user can save examples of both.
```

Comment:
It would be nice to highlight how this is different to [`filter processor`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) because I suspect that will be something that is asked.

Reply:
Perhaps some suggestions when this should be used instead of the filter processor.

Reply:
Thanks! great idea.

I added the following section to the README.md:

## Relation to Other Samplers
- [probabilisticsamplerprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor) will either keep the entire trace or drop the entire trace. It is not able to sample out just parts of a trace.
- [tailsamplingprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor) will either keep the entire trace or drop the entire trace. It is not able to sample out just parts of a trace.
- [filterprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor) can drop individual spans and keep others, but it is not aware of the position of the span in the trace and thus can remove non-leaf spans and break the trace into multiple fragments. Also, it is not a sampler and will always filter out matching spans without allowing any samples to be kept.

It is recommended to use the intracesampler for traces that are valuable to keep, but includes spans which are not.